### PR TITLE
getDisplayMedia for chrome

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: modules/RTC/ScreenObtainer.js [hunk 1]
```diff
@@ -167,11 +174,36 @@ const ScreenObtainer = {
                 logger.info('Chrome extension not supported until ver 34');
 
                 return null;
+            } else if (navigator.getDisplayMedia
```
[the remark this answers is not in the document]
Comment (bbaldino) [reply modules/RTC/ScreenObtainer.js]: and how about moving this getdisplaymedia logic to a helper function, since this if/else if tree is pretty big and a little hard to follow.

File: modules/RTC/ScreenObtainer.js [hunk 2]
```diff
@@ -167,11 +174,36 @@ const ScreenObtainer = {
...
+                            let applyConstraintsPromise;
+
+                            if (stream
+                                && stream.getTracks()
+                                && stream.getTracks().length > 0) {
+                                applyConstraintsPromise
+                                    = stream.getTracks()[0]
+                                        .applyConstraints(getSSConstraints());
+                            } else {
+                                applyConstraintsPromise = Promise.resolve();
```
Comment (bbaldino) [inline modules/RTC/ScreenObtainer.js]: so if we get nothing we just want to resolve cleanly?  i guess that must be what the other flows do?

File: modules/RTC/ScreenObtainer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (damencho) [reply modules/RTC/ScreenObtainer.js]: Will fix it.

File: modules/RTC/ScreenObtainer.js [hunk 3]
```diff
@@ -167,11 +174,36 @@ const ScreenObtainer = {
                 logger.info('Chrome extension not supported until ver 34');
 
                 return null;
+            } else if (navigator.getDisplayMedia
+                        && !options.desktopSharingChromeDisabled) {
+                return (ops, onSuccess, onFailure) => {
```
[the remark this answers is not in the document]
Comment (damencho) [reply modules/RTC/ScreenObtainer.js]: Ups, that one slipt into the commit. It is not used.

File: modules/RTC/ScreenObtainer.js [hunk 4]
```diff
@@ -167,11 +174,36 @@ const ScreenObtainer = {
...
+                    navigator.getDisplayMedia({ video: true })
+                        .then(stream => {
+                            let applyConstraintsPromise;
+
+                            if (stream
+                                && stream.getTracks()
+                                && stream.getTracks().length > 0) {
+                                applyConstraintsPromise
+                                    = stream.getTracks()[0]
+                                        .applyConstraints(getSSConstraints());
```
[the remark this answers is not in the document]
Comment (damencho) [reply modules/RTC/ScreenObtainer.js]: Yep, that one can come with the `ops` ^

File: modules/RTC/ScreenObtainer.js (hunk 2 above)
Comment (damencho) [reply modules/RTC/ScreenObtainer.js]: Well, current chrome checks for stream === '' which indicates a cancel came from chrome extension, otherwise, fail.
Where Firefox just resolves with whatever is received and the same is for Edge getDisplayMedia. So not sure what to do?
I've added the check just to be safe when doing stream.getTracks()[0].

File: modules/RTC/ScreenObtainer.js [hunk 5]
```diff
@@ -423,6 +418,37 @@ const ScreenObtainer = {
...
+    obtainScreenFromGetDisplayMedia(options, callback, errorCallback) {
+        navigator.getDisplayMedia({ video: true })
+            .then(stream => {
+                let applyConstraintsPromise;
+
+                if (stream
+                    && stream.getTracks()
+                    && stream.getTracks().length > 0) {
+                    applyConstraintsPromise = stream.getTracks()[0]
+                        .applyConstraints(options.trackOptions);
```
Comment (bbaldino) [inline modules/RTC/ScreenObtainer.js]: i don't think i saw why we have to do this after (apply the constraints) as opposed to setting these in the getDisplayMedia call, does the API not work that way?

Comment (damencho) [reply modules/RTC/ScreenObtainer.js]: No the, spec constraints that you can pass to getDisplayMedia are audio: true/false, video:true: false and some os specific about surface and cursor.
I was checking the issue tracker of the spec about the framerate and this is what they say should be done and also mentioned here: https://webrtchacks.com/chrome-screensharing-getdisplaymedia/
https://github.com/w3c/mediacapture-screen-share/issues/49
